doc/api/url.md

```diff
@@ -1562,16 +1565,9 @@ A `URIError` is thrown if the `auth` property is present but cannot be decoded.
 
 `url.parse()` uses a lenient, non-standard algorithm for parsing URL
 strings. It is prone to security issues such as [host name spoofing][]
-and incorrect handling of usernames and passwords.
-
-`url.parse()` is an exception to most of the legacy APIs. Despite its security
-concerns, it is legacy and not deprecated because it is:
-
-* Faster than the alternative WHATWG `URL` parser.
```

Comment:
Do we know which one is faster?

Reply:
I suspect @anonrig might be able to answer that. Generally, I'd say `url.parse()` is usually faster, but not necessarily by a whole lot in the typical use cases. And it is more susceptible to things like host spoofing than WHATWG `URL`. But anything @anonrig says to the contrary should be treated as the more informed opinion that it is.

Reply:
Url.parse is faster but not spec compliant.

Reply:
When parsing a relative URL, then there is no possibilliy of username/password and therefore no issue with the spec, right?

Reply:
I can’t answer that question with 100% uncertainity, and I think nobody can.

Reply:
Let me rephrase the question:

Has anyone reported bugs or vulnerabilities for relative urls as inputs to url.parse()?

Reply:
`url.parse()` used to treat the `\bhttp://example.com/\b` as a relative URL. WHATWG URL strips leading and trailing backspaces and treats it as a fully-qualified URL. So that's an example where `url.parse()` mishandling a URL and treating it like a relative URL could lead to problems. (This issue has been fixed.)

Reply:
I see. I'm thinking more along the lines of URLs that start with `/`, like req.url

 ```js
function handler(req, res) {
  if (req.url.startsWith('/')) {
    const parsed = url.parse(req.url)
    parsed.search = '?v=2'
    const result = url.format(parsed)
    res.end(`Modified url: ${result}`)
  } else {
    throw new Error('Expected req.url to be relative')
  }
}
```

This is the case for which `url.parse()` and `url.format()` is useful, and theres not a good solution today using `new URL()`.